src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionContextFactory.cs

```diff
@@ -48,6 +48,14 @@ public SocketConnectionContextFactory(SocketConnectionFactoryOptions options, IL
         var maxWriteBufferSize = _options.MaxWriteBufferSize ?? 0;
         var applicationScheduler = options.UnsafePreferInlineScheduling ? PipeScheduler.Inline : PipeScheduler.ThreadPool;
 
+        // Socket callbacks run on the threads polling for IO if we're using the old thread windows thread pool
+        var dispatchSocketCallbacks = OperatingSystem.IsWindows() &&
+                                      (Environment.GetEnvironmentVariable("DOTNET_ThreadPool_UsePortableThreadPoolForIO") == "0" ||
+                                      Environment.GetEnvironmentVariable("COMPlus_ThreadPool_UsePortableThreadPoolForIO") == "0");
```

Comment:
What if someone sets this to "false". Would the runtime interpret that differently than we are here? This really isn't an app context switch?

Reply:
The runtime only supports 1 and 0 for these AFAIK. 
cc @kouvel

Reply:
https://github.com/dotnet/runtime/blob/8f18a0dff889449ab62086134903f89c7325b13f/src/coreclr/inc/clrconfigvalues.h#L533

@jkotas @vitek-karas Is there a better way to read these settings from managed code?

Reply:
This is about as good as you can do (in some cases, the alternative is to use private reflection).

I assume that this is temporary hack for .NET 7 only. Correct?

Reply:
Correct!